src/Microsoft.AspNetCore.Routing/DefaultParameterPolicyFactory.cs

```diff
@@ -44,47 +44,19 @@ public override IParameterPolicy Create(RoutePatternParameterPart parameter, str
                 throw new ArgumentNullException(nameof(inlineText));
             }
 
-            // Example:
-            // {productId:regex(\d+)}
-            //
-            // ParameterName: productId
-            // value: regex(\d+)
-            // name: regex
-            // argument: \d+
-            (var name, var argument) = Parse(inlineText);
+            var parameterPolicy = ParameterPolicyActivator.ResolveParameterPolicy(_options.ConstraintMap, _serviceProvider, inlineText);
```

Comment:
`DefaultParameterPolicyFactory` is a thin wrapper around `ParameterPolicyActivator`. Avoid duplication between it and `DefaultInlineConstraintResolver`